docs/pages/docs/handbook/deploying-to-docker.mdx

```diff
@@ -0,0 +1,155 @@
+
+# Copy app source
+COPY . .
+
+EXPOSE 8080
+
+CMD [ "node", "apps/docs/server.js" ]
+```
+
+This will copy the root `package.json` and the and the root lockfile to the docker image. Then, it'll install dependencies, copy the app source and start the app.
+
+You should also create a `.dockerignore` file to prevent node_modules from being copied in with the app's source.
+
+```txt filename=".dockerignore"
+node_modules
+npm-debug.log
+```
+
+### The lockfile changes too often
+
+Docker is pretty smart about how it deploys your apps. Just like Turbo, it tries to do as [little work as possible](http://bitjudo.com/blog/2014/03/13/building-efficient-dockerfiles-node-dot-js/).
+
+In our Dockerfile's case, it will only run `npm install` if the files it has in its image are _different_ from the previous run. If not, it'll restore the `node_modules` directory it had before.
+
+This means that whenever `package.json`, `apps/docs/package.json` or `package-lock.json` change, the docker image will run `npm install`.
+
+This sounds great - until we realise something. The `package-lock.json` is _global_ for the monorepo. That means that **if we install a new package inside `apps/web`, we'll cause `apps/docs` to redeploy**.
+
+In a large monorepo, this can result in a huge amount of lost time, as any change to a monorepo's lockfile cascades into tens or hundreds of deploys.
+
+## The solution
+
+The solution is to prune the inputs to the Dockerfile to only what is strictly necessary. Turborepo provides a simple solution - `turbo prune`.
+
+```bash
+turbo prune --scope="docs" --docker
+```
+
+Running this command creates a **pruned version of your monorepo** inside an `./out` directory. It only includes workspaces which `docs` depends on.
+
+Crucially, it also **prunes the lockfile** so that only the relevant `node_modules` will be downloaded.
+
+### The `--docker` flag
+
+By default, `turbo prune` puts all relevant files inside `./out`. But to optimize caching with Docker, we ideally want to copy the files over in two stages.
+
+First, we want to copy over only what we need to install the packages. When running `--docker`, you'll find this inside `./out/json`.
+
+```txt
+out
+├── json
+│   └── apps
+│       └── docs
+│           └── package.json
+├── full
+│   └── apps
+│       └── docs
+│           ├── server.js
+│           └── package.json
+└── package-lock.json
+```
+
+Afterwards, you can copy the files in `./out/full` to add the source files.
+
+Splitting up **dependencies** and **source files** in this way lets us **only run `npm install` when dependencies change** - giving us a much larger speedup.
+
+<Callout type="info">
+  Without `--docker`, all pruned files are placed inside `./out`.
+</Callout>
+
+### Example
+
+Our detailed [`with-docker` example](https://github.com/vercel/turborepo/tree/main/examples/with-docker) goes into depth on how to utilise `prune` to its full potential. Here's the Dockerfile, copied over for convenience.
+
+```docker
```

Comment:
Could we add a comment [here](https://github.com/vercel/turborepo/blob/main/examples/with-docker/apps/api/Dockerfile) and [here](https://github.com/vercel/turborepo/blob/main/examples/with-docker/apps/web/Dockerfile) with something like:

This Dockerfile is included in docs - if updating, please update the docs as well (if relevant)

Reply:
Done!